src/components/popover-container/popover-container.component.tsx

```diff
@@ -15,34 +15,124 @@ import createGuid from "../../__internal__/utils/helpers/guid";
 import { filterStyledSystemPaddingProps } from "../../style/utils";
 import ClickAwayWrapper from "../../__internal__/click-away-wrapper";
 
-const paddingPropTypes = filterStyledSystemPaddingProps(
-  styledSystemPropTypes.space
+export interface RenderOpenProps {
+  tabIndex: number;
+  isOpen?: boolean;
+  "data-element": string;
+  onClick: (ev: React.MouseEvent<HTMLElement>) => void;
+  ref: React.RefObject<HTMLButtonElement>;
+  "aria-label"?: string;
+  id?: string;
+}
+
+const renderOpen = ({
+  tabIndex,
+  onClick,
+  "data-element": dataElement,
+  ref,
+  "aria-label": ariaLabel,
+  id,
+}: RenderOpenProps) => (
+  <PopoverContainerOpenIcon
+    tabIndex={tabIndex}
+    onAction={onClick}
+    data-element={dataElement}
+    ref={ref}
+    aria-label={ariaLabel}
+    aria-haspopup
+    id={id}
+  >
+    <Icon type="settings" />
+  </PopoverContainerOpenIcon>
+);
+
+export interface RenderCloseProps {
+  "data-element": string;
+  tabIndex: number;
+  onClick: (ev: React.MouseEvent<HTMLElement>) => void;
+  ref: React.RefObject<HTMLButtonElement>;
+  "aria-label": string;
+}
+
+const renderClose = ({
+  "data-element": dataElement,
+  tabIndex,
+  onClick,
+  ref,
+  "aria-label": ariaLabel,
+}: RenderCloseProps) => (
+  <PopoverContainerCloseIcon
+    data-element={dataElement}
+    tabIndex={tabIndex}
+    onAction={onClick}
+    ref={ref}
+    aria-label={ariaLabel}
+  >
+    <Icon type="close" />
+  </PopoverContainerCloseIcon>
 );
 
-const PopoverContainer = ({
+export interface PopoverContainerProps extends PaddingProps {
+  /** A function that will render the open component
+   *
+   * `({tabIndex, isOpen, data-element, onClick, ref, aria-label}) => ()`
+   *
+   */
+  renderOpenComponent?: (args: RenderOpenProps) => JSX.Element;
+  /** A function that will render the close component
+   *
+   * `({data-element, tabIndex, onClick, ref, aria-label}) => ()`
+   *
+   */
+  renderCloseComponent?: (args: RenderCloseProps) => JSX.Element;
+  /** The content of the popover-container */
+  children?: React.ReactNode;
+  /** Sets rendering position of dialog */
+  position?: "left" | "right";
+  /** Sets the popover container dialog header name */
+  title?: string;
+  /** Callback fires when close icon clicked */
+  onClose?: (ev: React.MouseEvent<HTMLElement>) => void;
```

Comment:
I could be wrong or suggesting something heretical here but setting the type like so would mean we don't need to use `any`
```suggestion
  onClose?: (ev: React.MouseEvent<HTMLElement> | CustomEvent) => void;
```